src/Collection.php

```diff
@@ -94,7 +96,13 @@ public static function range($start = 0, $end = null, $step = 1)
     public function getIterator()
     {
         if ($this->inputFactory) {
-            $this->input = call_user_func($this->inputFactory);
+            $input = call_user_func($this->inputFactory);
+
+            if (is_array($input)) {
+                $input = new RecursiveArrayIterator($input);
+            }
+
```

Comment:
The real question is why does it think that `$this->input` is not a Traversable at this point. Probably because we can't enforce the return type of the inputFactory. So it may be beneficial to throw here as the inputFactory may return Traversable on first call and string on second (by holding state somewhere).